src/rpcblockchain.cpp

```diff
@@ -1056,6 +1056,135 @@ UniValue rain(const UniValue& params, bool fHelp)
     return res;
 }
 
+UniValue rainbymagnitude(const UniValue& params, bool fHelp)
+{
+    if (fHelp || (params.size() < 1 || params.size() > 2))
+        throw runtime_error(
+                "rainbymagnitude <amount> [message]\n"
+                "\n"
+                "<amount> --> Required: Specify amount of coints in double to be rained\n"
+                "[message] -> Optional: Provide a message rained to all rainees\n"
+                "\n"
+                "rain coins by magnitude on network");
+
+    UniValue res(UniValue::VOBJ);
+
+    double dAmount = params[0].get_real();
+
+    if (dAmount <= 0)
+        throw runtime_error("Amount must be greater then 0");
+
+    std::string sMessage = "";
+
+    if (params.size() > 1)
+        sMessage = params[1].get_str();
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    CWalletTx wtx;
+    wtx.mapValue["comment"] = "Rain By Magnitude";
+    std::set<CBitcoinAddress> setAddress;
+    std::vector<std::pair<CScript, int64_t> > vecSend;
+
+    wtx.hashBoinc = "<NARR>Rain By Magnitude: " + MakeSafeMessage(sMessage) + "</NARR>";
+
+    // Gather Magnitude data
+    std::string sSuperblock = ReadCache("superblock", "magnitudes").value;
+
+    if (sSuperblock.empty())
+        throw runtime_error("Unable to load superblock data");
+
+    std::vector<std::string> vSuperblock = split(sSuperblock, ";");
+
+    double dTotalAmount = 0;
+    int64_t nTotalAmount = 0;
+
+    StructCPID structcpid = mvNetwork["NETWORK"];
+
+    // Use total network magnitude here as using 115000 can results in a lower then intended sent amount due to a missing project
+    double dTotalNetworkMagnitude = structcpid.NetworkMagnitude;
+
+    for (unsigned int i = 0; i < vSuperblock.size(); i++)
```

Comment:
Instead of going through the superblock you can check `mvMagnitudes` which should contain the same information.

Reply:
will do tmr